db/db.go

```diff
@@ -0,0 +1,457 @@
+/*
+Copyright 2025 Google Inc. All Rights Reserved.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Package db manages the googet state sqlite database.
+package db
+
+import (
+	"context"
+	"database/sql"
+	"errors"
+	"fmt"
+	"io/fs"
+	_ "modernc.org/sqlite" // Import the SQLite driver (unnamed)
+	"os"
+	"strconv"
+	"strings"
+
+	"github.com/google/googet/v2/client"
+	"github.com/google/googet/v2/goolib"
+)
+
+type gooDB struct {
+	db *sql.DB
+}
+
+// NewDB returns the googet DB object
+func NewDB(dbFile string) (*gooDB, error) {
+	var gdb gooDB
+	if _, err := os.Stat(dbFile); errors.Is(err, fs.ErrNotExist) {
+		gdb.db, _ = createDB(dbFile)
+		return &gdb, nil
+	}
+	goodb, err := sql.Open("sqlite", dbFile)
+	if err != nil {
+		return nil, err
+	}
+	gdb.db = goodb
+	return &gdb, nil
+}
+
+// Create db creates the initial googet database
+func createDB(dbFile string) (*sql.DB, error) {
+	goodb, err := sql.Open("sqlite", dbFile)
+	if err != nil {
+		return nil, err
+	}
+
+	createDBQuery := `BEGIN;
+	CREATE TABLE IF NOT EXISTS state (
+			id INTEGER PRIMARY KEY AUTOINCREMENT,
+			PkgName TEXT NOT NULL UNIQUE,
+			SourceRepo TEXT NOT NULL,
+			DownloadURL TEXT NOT NULL,
+			Checksum TEXT,
+			LocalPath TEXT,
+			UnpackDir TEXT
+		);
+	CREATE TABLE IF NOT EXISTS pkgspec (
+			id INTEGER PRIMARY KEY AUTOINCREMENT,
+			PkgName TEXT NOT NULL UNIQUE,
+			Version TEXT NOT NULL,
+			Arch TEXT,
+			Description TEXT,
+			License TEXT,
+			Authors TEXT,
+			Owners TEXT,
+			Source TEXT,
+			Replaces TEXT,
+			Conflicts TEXT
+		);
+	CREATE TABLE IF NOT EXISTS pkgInstallers (
+			id INTEGER PRIMARY KEY AUTOINCREMENT,
+			PkgName TEXT NOT NULL,
+			ScriptType TEXT NOT NULL,
+			Path TEXT NOT NULL,
+			Args TEXT,
+			ExitCodes TEXT,
+			UNIQUE(PkgName, ScriptType) ON CONFLICT REPLACE
+		);
+	CREATE TABLE IF NOT EXISTS pkgFiles (
+			id INTEGER PRIMARY KEY AUTOINCREMENT,
+			PkgName TEST NOT NULL,
+			FileName TEXT NOT NULL,
+			Path TEXT NOT NULL
+		);
+	CREATE TABLE IF NOT EXISTS pkgDeps (
+			id INTEGER PRIMARY KEY AUTOINCREMENT,
+			PkgName TEST NOT NULL,
+			Dependency TEXT NOT NULL,
+			Version TEXT
+		);
+	CREATE TABLE IF NOT EXISTS pkgInsFiles (
+			id INTEGER PRIMARY KEY AUTOINCREMENT,
+			PkgName TEST NOT NULL,
+			Name NOT NULL,
+			Hash NOT NULL
+		);
+	CREATE TABLE IF NOT EXISTS pkgTags (
+			id INTEGER PRIMARY KEY AUTOINCREMENT,
+			PkgName TEST NOT NULL,
+			Name NOT NULL,
```

Comment:
Assuming that both Name and Description here are meant to be TEXT as well instead of BLOB.